app/views/Project/Tagging/Sources/SourcesTable/LeadCopyModal/index.tsx

```diff
@@ -0,0 +1,242 @@
+                    {
+                        variant: 'success',
+                    },
+                );
+            },
+            onError: () => {
+                alert.show(
+                    'Failure copying leads',
+                    {
+                        variant: 'error',
+                    },
+                );
+            },
+        },
+    );
+
+    const projectDetailsVariables = useMemo(() => ({
+        ids: selectedProjects,
+    }), [selectedProjects]);
+
+    const {
+        loading: projectDetailPending,
+        data: projectDetailData,
+    } = useQuery<ProjectDetailsForLeadCopyQuery, ProjectDetailsForLeadCopyQueryVariables>(
+        PROJECT_DETAILS_FOR_LEAD_COPY,
+        {
+            variables: projectDetailsVariables,
+        },
+    );
+    const selectedUserProjects = projectDetailData?.projects?.results ?? [];
+    const selectedProjectsRendererParams = useCallback((_, datum) => ({
+        title: datum?.title,
+    }), []);
+
+    const handleCopyLeadsClick = useCallback(() => {
+        leadCopy({
+            variables: {
+                activeProjectId: projectId,
+                leadIds: leadId,
+                projectIds: selectedProjects,
+            },
+        });
+    }, [leadId, selectedProjects, leadCopy, projectId]);
+
+    return (
+        <Modal
+            className={styles.leadCopyModal}
+            onCloseButtonClick={onClose}
+            heading="Copy leads to projects"
+            footerActions={(
+                <ConfirmButton
+                    name="leads-copy"
+                    title="Export selected sources to selected projects"
+                    onConfirm={handleCopyLeadsClick}
+                    message={_cs(
+                        'Are you sure you want to copy ',
+                        leadId.length.toString(),
+                        'leads to ',
+                        selectedProjects.length.toString(),
+                        'projects?',
+                    )}
+                >
+                    Export
+                </ConfirmButton>
+            )}
+        >
+            <TextOutput
+                label="No. of leads to be copied"
+                value={leadId?.length}
+            />
+            <SearchMultiSelectInput
+                name="searchProjects"
+                value={selectedProjects}
+                className={styles.searchInput}
+                keySelector={searchProjectsKeySelector}
+                labelSelector={searchProjectsLabelSelector}
+                onChange={setSelectedProjects}
+                onSearchValueChange={setSearchText}
+                options={projectOptions}
+                searchOptions={projectOptions}
+                onOptionsChange={setProjectOptions}
+                optionsPending={loading}
+                totalOptionsCount={data?.projects?.totalCount ?? undefined}
+                onShowDropdownChange={setOpened}
+            />
```

Comment:
Why not use a ProjectSearchSelectInput component? Let's modify existing or create a new one.